playbooks/common/openshift-cluster/upgrades/rpm_upgrade.yml

```diff
@@ -3,24 +3,35 @@
 
 # Master package upgrade ends up depending on node and sdn packages, we need to be explicit
 # with all versions to avoid yum from accidentally jumping to something newer than intended:
+
 - name: Upgrade master packages
-  package: name={{ item }} state=present
+  package: name={{ master_pkgs | join(',') }} state=present
+  vars:
+    master_pkgs:
+      - "{{ openshift.common.service_type }}{{ openshift_pkg_version }}"
+      - "{{ openshift.common.service_type }}-master{{ openshift_pkg_version }}"
+      - "{{ openshift.common.service_type }}-node{{ openshift_pkg_version }}"
+      - "{{ openshift.common.service_type }}-sdn-ovs{{ openshift_pkg_version }} if openshift_use_openshift_sdn | default(true) else omit"
```

Comment:
You may have to do something like this since `omit` will add a string to this list.

```yaml
- name: Upgrade master packages
  package: name="{{ master_pkgs | difference(['remove_me']) | join(',') }}" state=present
  vars:
    master_pkgs:
    - "{{ openshift.common.service_type }}{{ openshift_pkg_version }}"
    - "{{ openshift.common.service_type }}-master{{ openshift_pkg_version }}"
    - "{{ openshift.common.service_type }}-node{{ openshift_pkg_version }}"
    - "{{ openshift.common.service_type }}-sdn-ovs{{ openshift_pkg_version }} if openshift_use_openshift_sdn | default(true) else 'remove_me'"
```